wordpress-mobile/WordPress-iOS
# 5245-Implement WordPress.com Rest API Using NSURLSession on MediaServiceRemoteRest

Refs #5245

This is the first step to implementing our Network layer using NSURLSession stack.
In this PR I implemented a first version of the WP.com REST API object using the AFNetworking class that use NSURLSession.

One of the main design decisions was to hide away the implementations details on how the Network layer is implemented in a way that we can later switch to another solution if we want to.

This PR only changed the MediaServiceRest to use the new API object other PRs will follow that will change the other services one by one.

To test:

 - Start  a new post on a WP.com blog/site
 - Add media to it image and videos.
 - See if the progress indicator are working.
 - See if it's possible to cancel the uploads
 - Check setting an featured image
 - When inserting media check if it's possible to view the WordPress media library.
 - Do the same tests using the legacy editor.

Needs review: @koke
